### ISSUE TITLE
epg_url does not work with gzip 

### ISSUE THREAD

**Issue by blockloop** [euzu/m3u-filter#62]
**Describe the bug**
Many providers send epg with gzip because of the sheer size. When I set the epg_url to my provider that provides gzip I get 

```
iptv-m3u-filter-1  | thread 'main' panicked at src/processing/xmltv_parser.rs:59:41:
iptv-m3u-filter-1  | called `Result::unwrap()` on an `Err` value: EscapeError(UnterminatedEntity(654..1148))
iptv-m3u-filter-1  | stack backtrace:
iptv-m3u-filter-1  |    0:     0x7fa302b3aba1 - std::backtrace_rs::backtrace::libunwind::trace::h103e47e00552e141
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/../../backtrace/src/backtrace/libunwind.rs:105:5
iptv-m3u-filter-1  |    1:     0x7fa302b3aba1 - std::backtrace_rs::backtrace::trace_unsynchronized::h592221613f1fd095
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
iptv-m3u-filter-1  |    2:     0x7fa302b3aba1 - std::sys_common::backtrace::_print_fmt::hf8a30f8299b1bef3
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:68:5
iptv-m3u-filter-1  |    3:     0x7fa302b3aba1 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h5a6e3e7cc140e753
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:44:22
iptv-m3u-filter-1  |    4:     0x7fa30297143c - core::fmt::rt::Argument::fmt::hd62a89729b8d4079
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/fmt/rt.rs:142:9
iptv-m3u-filter-1  |    5:     0x7fa30297143c - core::fmt::write::haa8a32127a7b0d75
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/fmt/mod.rs:1153:17
iptv-m3u-filter-1  |    6:     0x7fa302b1309e - std::io::Write::write_fmt::hc541adb06d0aeac1
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/io/mod.rs:1843:15
iptv-m3u-filter-1  |    7:     0x7fa302b3fc4f - std::sys_common::backtrace::_print::h59dc0d5369ede572
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:47:5
iptv-m3u-filter-1  |    8:     0x7fa302b3fc4f - std::sys_common::backtrace::print::hbb7b999054d983a7
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:34:9
iptv-m3u-filter-1  |    9:     0x7fa302b3f55d - std::panicking::default_hook::{{closure}}::hd143a60ad2fc5a07
iptv-m3u-filter-1  |   10:     0x7fa302b40470 - std::panicking::default_hook::h2aff2dd57e84e09f
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:292:9
iptv-m3u-filter-1  |   11:     0x7fa302b40470 - std::panicking::rust_panic_with_hook::h6811313e71f57da8
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:779:13
iptv-m3u-filter-1  |   12:     0x7fa302b3ffb0 - std::panicking::begin_panic_handler::{{closure}}::h2eb4991eec554504
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:657:13
iptv-m3u-filter-1  |   13:     0x7fa302b3ff06 - std::sys_common::backtrace::__rust_end_short_backtrace::h05983ae26e2c8878
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:171:18
iptv-m3u-filter-1  |   14:     0x7fa302b3fef3 - rust_begin_unwind
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
iptv-m3u-filter-1  |   15:     0x7fa3027cf074 - core::panicking::panic_fmt::h49eed7d6e9368ad2
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
iptv-m3u-filter-1  |   16:     0x7fa3027cf492 - core::result::unwrap_failed::hb815ab7de92baab5
iptv-m3u-filter-1  |                                at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/result.rs:1654:5
iptv-m3u-filter-1  |   17:     0x7fa3028b4220 - m3u_filter::processing::playlist_processor::process_source::{{closure}}::h94217eee680d13a0
iptv-m3u-filter-1  |   18:     0x7fa3028bbf70 - m3u_filter::processing::playlist_processor::exec_processing::{{closure}}::hdf6b1d154131ce1c
iptv-m3u-filter-1  |   19:     0x7fa30283756d - tokio::runtime::task::raw::poll::h1b85bec42757be42
iptv-m3u-filter-1  |   20:     0x7fa302b48cc9 - tokio::task::local::LocalSet::tick::h9177afd50173e071
iptv-m3u-filter-1  |   21:     0x7fa3028363f9 - tokio::task::local::LocalSet::run_until::{{closure}}::h15f104b6cbb93897
iptv-m3u-filter-1  |   22:     0x7fa3028d2e47 - m3u_filter::main::hb7eae3e9f4c84f8b
iptv-m3u-filter-1  |   23:     0x7fa30281905c - std::sys_common::backtrace::__rust_begin_short_backtrace::h10fb5c891061533a
iptv-m3u-filter-1  |   24:     0x7fa3028dc920 - main

```

**Comment by euzu** [comment]
@blockloop i have pushed the changes to the master, please give me a feedback after testing

**Comment by euzu** [comment]
closing due inactivity, should be solved with v2.0.2
